### PR TITLE
[5.1] Validator now accepts short types

### LINKED ISSUE #10468: Validation to accept short types
Validator doesn't accept short types, but Eloquent casting does.

### PR DESCRIPTION
**Pull request by jbrooksuk**
Closes #10468

### CODE REVIEW

File: src/Illuminate/Validation/Validator.php [hunk 1]
```diff
@@ -2596,6 +2598,25 @@ protected function requireParameterCount($count, $parameters, $rule)
     }
 
     /**
+     * Normalizes a rule so that we can accept short types.
+     *
+     * @param string $rule
+     *
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: cs

File: src/Illuminate/Validation/Validator.php [hunk 2]
```diff
@@ -2596,6 +2598,25 @@ protected function requireParameterCount($count, $parameters, $rule)
     }
 
     /**
+     * Normalizes a rule so that we can accept short types.
+     *
+     * @param string $rule
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: cs

**Comment by GrahamCampbell** [comment]
You might need to add "Int" to the numericRules property too.

File: src/Illuminate/Validation/Validator.php (hunk 1 above)
Comment (jbrooksuk) [inline src/Illuminate/Validation/Validator.php]: You'd think I'd know by now ;)

**Comment by jbrooksuk** [comment]
Ah good idea!

**Comment by jbrooksuk** [comment]
All done!

File: src/Illuminate/Validation/Validator.php [hunk 3]
```diff
@@ -2596,6 +2598,24 @@ protected function requireParameterCount($count, $parameters, $rule)
     }
 
     /**
+     * Normalizes a rule so that we can accept short types.
+     *
+     * @param  string $rule
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: still missing a space here ;)

**Comment by GrahamCampbell** [comment]
:+1: I'd like to see this very much. :)

File: src/Illuminate/Validation/Validator.php (hunk 3 above)
Comment (jbrooksuk) [inline src/Illuminate/Validation/Validator.php]: Before the `string` type?

Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: no, before the dollar sign

Comment (jbrooksuk) [inline src/Illuminate/Validation/Validator.php]: Ah yeah.

**Comment by jbrooksuk** [comment]
All done :)

**Comment by GrahamCampbell** [comment]
I wonder if we should deprecate the long versions in 5.2, and remove in 5.3?

**Pull request merged**